llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -5041,6 +5041,19 @@ SDValue AMDGPUTargetLowering::PerformDAGCombine(SDNode *N,
     return performAssertSZExtCombine(N, DCI);
   case ISD::INTRINSIC_WO_CHAIN:
     return performIntrinsicWOChainCombine(N, DCI);
+  case AMDGPUISD::FMAD_FTZ: {
+    SDValue N0 = N->getOperand(0);
+    SDValue N1 = N->getOperand(1);
+    SDValue N2 = N->getOperand(2);
+    EVT VT = N->getValueType(0);
+
+    // FMAD_FTZ is a FMAD, but flushing to zero is allowed (not required).
```

Comment:
No, this is the node matching the instruction behavior. Just do the fold and flush it yourself

Reply:
So we do a mul, flush denormals to zero, and then the add, and flush again? (Since FMAD is two separate rounding steps) or does the flush only happen at the end?
Do we have a helper somewhere to flush a denormal to zero?

Reply:
I'd assume the intermediate does a flush but I'm not sure. I don't think there's a utility for that, but it's simple to add

Reply:
Also there's a flush of the input and the output